mapproxy/config/loader.py

```diff
@@ -2327,12 +2327,55 @@ def merge_dict(conf, base):
             base[k] = v
         else:
             if isinstance(base[k], dict):
-                merge_dict(v, base[k])
+                if v != None:
+                    base[k] = merge_dict(v, base[k])
+            elif isinstance(base[k], list):
+                if v != None:
+                    if k in ['bbox', 'tile_size', 'max_output_pixels', 'sources', 'grids']:
+                        base[k] = v
+                    elif k in ['layers']:
+                        base[k] = merge_layers(v, base[k])
+                    elif len(v) == 0:  # delete
+                        base[k] = None
+                    else:
+                        base[k] = base[k] + v
             else:
+                # if k in ['layers']:
+                #     # todo: merge list of layer enumeration?
```

Comment:
Is it possible to resolve this TODO now?

Reply:
What do You mean by "if this make sense"?
If this path is not possible then there is no need to leave TODO
If path is possible but there is no implementation, then please throw error respectively.

Reply:
I have dug further and I found no case where this could be needed. I revert, rebase and pushed.